mesonbuild/dependencies/boost.py

```diff
@@ -688,8 +689,12 @@ def detect_roots(self) -> None:
             tmp: T.List[Path] = []
 
             # Add some default system paths
+            if m.is_darwin():
+                tmp.extend([
+                    Path('/opt/homebrew/'),        # for Apple Silicon MacOS
+                    Path('/usr/local/opt/boost'),  # for Intel Silicon MacOS
+                ])
             tmp += [Path('/opt/local')]
-            tmp += [Path('/usr/local/opt/boost')]
```

Comment:
This path is specific to Apple? Good to know...

Reply:
Yeah, homebrew on Intel puts its stuff there. I don't think it's a place we should necessary search automatically. I'm a little skeptical of some of the paths in there in general. Handling homebrew seems reasonable as it's pretty much de facto for Apple